plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -173,50 +174,14 @@ func (k *KafkaConsumer) Init() error {
 	}
 
 	config := sarama.NewConfig()
-	config.Consumer.Return.Errors = true
 
 	// Kafka version 0.10.2.0 is required for consumer groups.
 	config.Version = sarama.V0_10_2_0
 
-	if k.Version != "" {
-		version, err := sarama.ParseKafkaVersion(k.Version)
-		if err != nil {
-			return err
-		}
-
-		config.Version = version
-	}
-
-	if k.EnableTLS != nil && *k.EnableTLS {
-		config.Net.TLS.Enable = true
-	}
-
-	tlsConfig, err := k.ClientConfig.TLSConfig()
-	if err != nil {
+	if err := k.SetConfig(config); err != nil {
 		return err
 	}
 
-	if tlsConfig != nil {
-		config.Net.TLS.Config = tlsConfig
-
-		// To maintain backwards compatibility, if the enable_tls option is not
-		// set TLS is enabled if a non-default TLS config is used.
-		if k.EnableTLS == nil {
-			k.Log.Warnf("Use of deprecated configuration: enable_tls should be set when using TLS")
```

Comment:
After moving the tls config into common/kafka/config.go is there still a warning if someone uses enable_tls?

Reply:
no, it's just ignored. the setting doesn't make much sense. instead of you set any of the TLS settings it's automatically true internally

Reply:
If someone sets the tls settings to enable it but they have enable_tls = false, it could be confusing without a warning.  Similarly if they don't enable tls settings and they have enable_tls = true they might think they're getting tls when enable_tls is just being ignored.

It would make sense to me to add a warning if enable_tls is ever used.